api/v1/submissions/PKPSubmissionFileHandler.inc.php

```diff
@@ -0,0 +1,455 @@
+					'roles' => [ROLE_ID_MANAGER, ROLE_ID_SUB_EDITOR, ROLE_ID_ASSISTANT, ROLE_ID_AUTHOR],
+				],
+			],
+			'POST' => [
+				[
+					'pattern' => $this->getEndpointPattern(),
+					'handler' => [$this, 'add'],
+					'roles' => [ROLE_ID_MANAGER, ROLE_ID_SUB_EDITOR, ROLE_ID_ASSISTANT, ROLE_ID_AUTHOR],
+				],
+			],
+			'PUT' => [
+				[
+					'pattern' => $this->getEndpointPattern() . '/{submissionFileId}',
+					'handler' => [$this, 'edit'],
+					'roles' => [ROLE_ID_MANAGER, ROLE_ID_SUB_EDITOR, ROLE_ID_ASSISTANT, ROLE_ID_AUTHOR],
+				],
+			],
+			'DELETE' => [
+				[
+					'pattern' => $this->getEndpointPattern() . '/{submissionFileId}',
+					'handler' => [$this, 'delete'],
+					'roles' => [ROLE_ID_MANAGER, ROLE_ID_SUB_EDITOR, ROLE_ID_ASSISTANT, ROLE_ID_AUTHOR],
+				],
+			],
+		];
+		parent::__construct();
+	}
+
+	//
+	// Implement methods from PKPHandler
+	//
+	function authorize($request, &$args, $roleAssignments) {
+		$route = $this->getSlimRequest()->getAttribute('route');
+		import('lib.pkp.classes.security.authorization.SubmissionFileAccessPolicy'); // SUBMISSION_FILE_ACCESS_
+
+		import('lib.pkp.classes.security.authorization.ContextAccessPolicy');
+		$this->addPolicy(new ContextAccessPolicy($request, $roleAssignments));
+
+		import('lib.pkp.classes.security.authorization.SubmissionAccessPolicy');
+		$this->addPolicy(new SubmissionAccessPolicy($request, $args, $roleAssignments));
+
+		if ($route->getName() === 'add') {
+			$params = $this->getSlimRequest()->getParsedBody();
+			$fileStage = isset($params['fileStage']) ? (int) $params['fileStage'] : 0;
+			import('lib.pkp.classes.security.authorization.internal.SubmissionFileStageAccessPolicy');
+			$this->addPolicy(new SubmissionFileStageAccessPolicy($fileStage, SUBMISSION_FILE_ACCESS_MODIFY, 'api.submissionFiles.403.unauthorizedFileStageIdWrite'));
+
+		} elseif ($route->getName() === 'getMany') {
+			// Anyone passing SubmissionAccessPolicy is allowed to access getMany,
+			// but the endpoint will return different files depending on the user's
+			// stage assignments.
+
+		} else {
+			$accessMode = $this->getSlimRequest()->getMethod() === 'GET'
+				? SUBMISSION_FILE_ACCESS_READ
+				: SUBMISSION_FILE_ACCESS_MODIFY;
+			import('lib.pkp.classes.security.authorization.SubmissionFileAccessPolicy');
+			$this->addPolicy(new SubmissionFileAccessPolicy($request, $args, $roleAssignments, $accessMode, (int) $route->getArgument('submissionFileId')));
+		}
+
+		return parent::authorize($request, $args, $roleAssignments);
+	}
+
+	/**
+	 * Get a collection of submission files
+	 *
+	 * @param \Slim\Http\Request $slimRequest
+	 * @param APIResponse $response
+	 * @param array $args arguments
+	 * @return Response
+	 */
+	public function getMany($slimRequest, $response, $args) {
+		$request = $this->getRequest();
+
+		$params = [];
+
+		foreach ($slimRequest->getQueryParams() as $param => $val) {
+			switch ($param) {
+				case 'fileStages':
+				case 'reviewRoundIds':
```

Comment:
I think this entire case could be more simply written:
```
case 'reviewRoundIds':
    $params[$param] = array_map('intval', explode(',', $val));
    break;
```

Reply:
Nope, I'm wrong. But anyway, the `strpos($val, ',')` is unnecessary -- running a string without a delimiter in it through `explode` will happily give you an array anyway.

Reply:
Ah good. Cleans up a bunch of handlers with that.